src/messages/MClientRequest.h

```diff
@@ -252,10 +254,20 @@ class MClientRequest final : public MMDSOp {
   void encode_payload(uint64_t features) override {
     using ceph::encode;
     head.num_releases = releases.size();
-    head.version = CEPH_MDS_REQUEST_HEAD_VERSION;
+    /*
+     * If the peer is old version, we must skip all the
+     * new members, because the old version of MDS or
+     * client will just copy the 'head' memory and isn't
+     * that smart to skip them.
+     */
+    if (peer_old_version) {
+      head.version = 1;
+    } else {
+      head.version = CEPH_MDS_REQUEST_HEAD_VERSION;
+    }
 
     if (features & CEPH_FEATURE_FS_BTIME) {
-      encode(head, payload);
+      encode(head, payload, peer_old_version);
```

Comment:
(reposting my comment - not sure why github didn't publish)

I might be missing something obvious, but where is `peer_old_version` getting encoded?

Reply:
The `peer_old_version` won't be encoded, and it just a parameter for `struct ceph_mds_request_head:: encode()`. More detail please see the above code.

Reply:
Right. For some reason the diff render did not show 

```
void inline encode(const struct ceph_mds_request_head& h, ceph::buffer::list& bl, bool old_version)  {
```

... insanely which confused me :(

I can see it now. Thx!